types/mux.d.ts

```diff
@@ -0,0 +1,317 @@
+import { MetricsComparisonQueryParams } from './interfaces/MetricsComparisonQueryParams';
+import { MetricsComparisonValue } from './interfaces/MetricsComparisonValue';
+import { MetricsInsightsQueryParams } from './interfaces/MetricsInsightsQueryParams';
+import { MetricsOverallQueryParams } from './interfaces/MetricsOverallQueryParams';
+import { MetricsOverallValue } from './interfaces/MetricsOverallValue';
+import { MetricsTimeseriesQueryParams } from './interfaces/MetricsTimeseriesQueryParams';
+import { PlaybackId } from './interfaces/PlaybackId';
+import { PlaybackIdPolicy } from './interfaces/PlaybackIdPolicy';
+import { RealTimeBreakdownQueryParams } from './interfaces/RealTimeBreakdownQueryParams';
+import { RealTimeBreakdownResponse } from './interfaces/RealTimeBreakdownResponse';
+import { RealTimeDimensionsResponse } from './interfaces/RealTimeDimensionsResponse';
+import { RealTimeHistogramQueryParams } from './interfaces/RealTimeHistogramQueryParams';
+import { RealTimeHistogramResponse } from './interfaces/RealTimeHistogramResponse';
+import { RealTimeMetricsResponse } from './interfaces/RealTimeMetricsResponse';
+import { RealTimeTimeseriesParams } from './interfaces/RealTimeTimeseriesParams';
+import { RealTimeTimeseriesResponse } from './interfaces/RealTimeTimeseriesResponse';
+import { RequestOptions } from './interfaces/RequestOptions';
+import { SimulcastTarget } from './interfaces/SimulcastTarget';
+import { SimulcastTargetParams } from './interfaces/SimulcastTargetParams';
+import { StaticRenditions } from './interfaces/StaticRenditions';
+import { TextTrack } from './interfaces/TextTrack';
+import { Track } from './interfaces/Track';
+import { Upload } from './interfaces/Upload';
+import { VideoTrack } from './interfaces/VideoTrack';
+import { VideoView } from './interfaces/VideoView';
+import { VideoViewsQueryParams } from './interfaces/VideoViewsQueryParams';
+import { ViewError } from './interfaces/ViewError';
+
+export declare class Base extends EventEmitter {
+  private _tokenId?: string;
+  private _tokenSecret?: string;
+  private _config?: RequestOptions;
+  private _requestOptions?: RequestOptions;
+
+  constructor(base: Base | RequestOptions);
+  constructor(tokenId: string, tokenSecret: string, config: RequestOptions);
+  constructor(
+    param?: Base | RequestOptions | string,
+    tokenSecret?: string,
+    config?: RequestOptions
+  );
+  constructor(...params: any[]);
+}
+
+export declare interface ListParams {
+  limit?: number;
+  page?: number;
+}
+
+export declare interface CreatePlaybackIdParams {
+  policy: PlaybackIdPolicy;
+}
+
+export declare interface UpdateMp4SupportParams {
+  mp4_support: AssetMp4Support;
+}
+
+export declare interface UpdateMasterAccessParams {
+  master_access: AssetMasterAccess;
+}
+
+export declare class Assets extends Base {
+  create(params: CreateAssetParams): Promise<Asset>;
+  del(assetId: string): Promise<any>;
```

Comment:
If we're going to a major version, we may want to remove the `remove` function in `base.js`:

```
  remove(...params) {
    // eslint-disable-next-line no-console
    console.warn(
      'The remove helper has been deprecated in favor of del. `remove` will no longer be available after the next major version bump (3.0).'
    );
    return this.del(...params);
  }
```

Reply:
good call